config-scalecube-configuration-service/src/main/java/io/scalecube/config/service/ScalecubeConfigurationServiceConfigSource.java

```diff
@@ -0,0 +1,124 @@
+
+  ConfigurationService service;
+
+  private EntriesRequest requestEntries;
+
+  private final Parsing parsing = new Parsing();
+
+  private static URL defaultUrl() {
+    try {
+      return new URL("https", "configuration-service-http.genesis.om2.com", 443, "/");
+    } catch (MalformedURLException ignoredException) {
+      return null;
+    }
+  }
+
+  /**
+   * Create a configuration source that connects to the production environment of scalecube
+   * configuration service.
+   *
+   * @param token the API token
+   * @param repository the name of the repository
+   */
+  public ScalecubeConfigurationServiceConfigSource(String token, String repository) {
+    this(token, repository, defaultUrl());
+  }
+
+  /**
+   * Create a configuration source that connects to the production environment of scalecube
+   * configuration service.
+   *
+   * @param token the API token
+   * @param repository the name of the repository
+   * @param service the URL of this service
+   */
+  public ScalecubeConfigurationServiceConfigSource(String token, String repository, URL service) {
+    Builder builder =
+        builder()
+            .host(service.getHost())
+            .port(service.getPort())
+            .contentType(JacksonCodec.CONTENT_TYPE)
+            .loopResources(HttpResources.get());
+    if ("https".equals(service.getProtocol()) || "wss".equals(service.getProtocol())) {
+      builder = builder.secure();
+    }
+    this.service = http(builder.build()).forService(ConfigurationService.class);
+    requestEntries = new EntriesRequest(token, repository);
+  }
+
+  /**
+   * Create a configuration source that connects to the production environment of scalecube
+   * configuration service.
+   *
+   * @param token the API token
+   * @param repository the name of the repository
+   * @param service the actual configuration service
+   */
+  public ScalecubeConfigurationServiceConfigSource(
+      String token, String repository, ConfigurationService service) {
+    this.service = service;
+    requestEntries = new EntriesRequest(token, repository);
+  }
+
+  @Override
+  public Map<String, ConfigProperty> loadConfig() {
+    try {
+      return service
+          .entries(requestEntries)
+          .flatMapIterable(Function.identity())
+          .collectMap(FetchResponse::key, this.parsing::fromFetchResponse)
+          .block();
+    } catch (Exception e) {
+      e.printStackTrace();
+      LOGGER.warn("unable to load config properties", e);
+      throw new ConfigSourceNotAvailableException(e);
+    }
+  }
+
+  private static class Parsing {
+    private ObjectWriter writer;
+
+    protected Parsing() {
+      writer = ObjectMapperHolder.getInstance().writer(new MinimalPrettyPrinter());
+    }
+
+    public ConfigProperty fromFetchResponse(FetchResponse fetchResponse) {
+      try {
+        return LoadedConfigProperty.withNameAndValue(
+                fetchResponse.key(), writer.writeValueAsString(fetchResponse.value()))
```

Comment:
So just to understand we retrieve the `FetchResponse` and the Jackson turns value into `HashMap` structure, then we turn back this `HashMap` into `String` to put in the ConfigSource. Is that really?

Reply:
yes